x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -54,67 +55,91 @@ export const generateMlInferencePipelineBody = ({
   model,
   pipelineName,
 }: MlInferencePipelineParams): MlInferencePipeline => {
-  // if model returned no input field, insert a placeholder
-  const modelInputField =
-    model.input?.field_names?.length > 0 ? model.input.field_names[0] : 'MODEL_INPUT_FIELD';
-
-  // For now this only works for a single field mapping
-  const sourceField = fieldMappings[0].sourceField;
-  const targetField = fieldMappings[0].targetField;
   const inferenceType = Object.keys(model.inference_config)[0];
-  const remove = getRemoveProcessorForInferenceType(targetField, inferenceType);
-  const set = getSetProcessorForInferenceType(targetField, inferenceType);
-
-  return {
+  const pipelineDefinition: MlInferencePipeline = {
     description: description ?? '',
-    processors: [
-      {
-        remove: {
-          field: getMlInferencePrefixedFieldName(targetField),
-          ignore_missing: true,
-        },
+    processors: [],
+    version: 1,
+  };
+
+  // Add remove and inference processors
+  fieldMappings.forEach(({ sourceField, targetField }) => {
```

Comment:
This code adds support for generating a pipeline from multiple field mappings. I can split this into a separate PR to reduce the scope of this one, if needed.